LiteCore/Query/SQLiteN1QLFunctions.cc

```diff
@@ -813,28 +837,223 @@ namespace litecore {
 
 #pragma mark - DATES:
 
+    inline int64_t getQuarter(const DateTime& t) {
+        return (t.M + 2) / 3;
+    }
+
+    inline int64_t leapYearsWithin(int64_t year) {
```

Comment:
What's the source of these algorithms/formulas? Did they come from existing working code?

Reply:
I think the source was the N1QL go code from the server team.